rfcs/0000-kuadrant_crd.md

```diff
@@ -0,0 +1,164 @@
+
+Not every use cases requires, or even would want to permit usage of all of Kuadrant's capabilities to
+be enabled on the target cluster. This proposal aims at giving users control, through the `kuadrant`
+custom resource, of what gets enabled on the cluster when deploying Kuadrant.
+
+# Motivation
+[motivation]: #motivation
+
+The rational behind this proposal is that not every deployment requires all of Kuadrant features to be
+supported; currently `RateLimitPolicy` and `AuthPolicy`. There is no reason to impose the workload to
+run on the cluster if it isn't need, but there might be more reasons why explicitly disabling one
+feature might be desirable: disabling the users to actually apply one or the other policy to their
+traffic. Authorization for instance could always be implemented at the application level as per the
+organization desire, while rate limiting is achieved through `RateLimitPolicy` attached to the network
+resources. Just as the opposite situation might be applicable to an organization that wants to do
+authn/z by leveraging Kuadrant, but have no use cases for rate limiting.
+
+This proposal aims not only at defining a way to disable one of these features at deploy time, but
+also clarify what the behavior would be when such a state transitions while some policy are already
+existent; what the user experience would be when trying to apply a policy that's been explicitly
+disabled by the cluster administrator when Kuadrant was installed.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When applying the `Kuadrant` resource to the cluster for the first time, you have the option to
+disable the support for a policy by specifying the following in the `spec` of the resource:
+
+```yaml
+authentication:
+  policy: disabled
+rateLimiting:
+  policy: disabled
+```
+
+The `policy` key of either of the service, `authentication` and `rateLimiting`, defaults to `enabled`.
+That value can be specified as `disabled`. The behavior on both services is consistent: it disables
+the support for the policy backed by the service, i.e. `AuthPolicy` and `RateLimitPolicy`
+respectively.
+
+When disabling `RateLimitPolicy`, Kuadrant will not create the `Limitador` object used by the
+`limitador-operator`. Disabling the `AuthPolicy` yields the same result for the `Authorino` object.
+Additionally it will avoid registering the [external
+authorizer](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/) with Istio. The
+both policies API will be exposed and the reconcilers will still be registered. Creating a policy that
+is disabled would be denied.
+
+A disabled policy can be enabled by changing the value of the `policy` key. Flipping the value can
+only happen when no policy object exist, e.g. in order to disable the `authentication`, no
+`AuthPolicy` can exist or this would fail.
+
+While no default `Limitador` or `Authorino` object would be created, a user can absolutely create one
```

Comment:
Would this also hold true for AuthPolicy and RateLimitPolicy. IE the CRDs would still be present and you could still create one but it would not be acted on? If so does this break the previous rule that it can't be disabled if a policy exists?

Reply:
I see you have coverd this below with the validating admission webhook